app/models/exercise.rb

```diff
@@ -70,6 +70,10 @@ def self.for(track_slug, exercise_slug)
     end
   end
 
+  after_commit do
+    track.recache_num_exercises! if (saved_changes.keys & %w[id status]).present?
```

Comment:
Interesting. Your using the fact the `id` column changes if the record is deleted, right?

Reply:
Yup!